New fix flaky

akib6075 opened the pull request:
Fixes #ISSUSE_ID
JSONUtil.toJSONString() which is coming from JacksonHandler returning unordered object which fails in test.
So I changed the testcase as well.
Please have a look.
Flaky test evidence using the NonDex tool: https://github.com/akib6075/hippo4j/actions/runs/4519255268
after resolving flaky test evidence using the NonDex tool: https://github.com/akib6075/hippo4j/actions/runs/4519274552
Changes proposed in this pull request:
- 

> Check mailbox configuration when submitting. [Contributor Guide](https://hippo4j.cn/community/contributor-guide)

akib6075 committed:
Create new_test_main.yml

akib6075 committed:
Create new_test_main.yml

KaziAkibJaved committed:
potentially issue fixed

KaziAkibJaved committed:
potentially issue fixed

dmego committed:
optimize: fix StyleCheck errors for hippo4j-threadpool-core (#1289)

* optimize: fix StyleCheck errors for hippo4j-threadpool-core

* add TaskTimeRecordPlugin#TableSizeFor test method

dmego committed:
optimize: fix StyleCheck errors for hippo4j (#1291)

* optimize: fix StyleCheck errors for hippo4j

* ignore TypeName error

magestacks committed:
Public module removes spring dependency (#1293)

DDreame committed:
optimize: fix StyleCheck errors for hippo4j-agent-bootstrap (#1292)

* optimize: fix StyleCheck errors for hippo4j-agent-bootstrap

* optimize: fix StyleCheck errors for hippo4j-agent-bootstrap

* optimize: fix StyleCheck errors for hippo4j-agent-bootstrap

yanrongzhen committed:
Implement real-time parameter modification of the Spring bean thread pool instance. (#1295)

* Add common dependencies and realize basic functions.

* Update example config properties

DDreame committed:
optimize: fix StyleCheck errors for hippo4j-threadpool-coore (#1294)

* optimize: fix StyleCheck errors for hippo4j-threadpool-coore

* optimize: fix StyleCheck errors for hippo4j-threadpool-coore

magestacks committed:
Update user company (#1298)

* Update user company

* Update user company

Mryanhehe committed:
update property file about zookeeper example (#1302)

Co-authored-by: Yixuan yan <yixuanyan@YixuandeMacBook-Air.local>

magestacks committed:
Abstract infrastructure modules and refactor existing modules (#1305)

yanrongzhen committed:
Refactor agent module directory structure. (#1306)

pizihao committed:
Refactor the rpc module (#1309)

magestacks committed:
Registered company (#1311)

lucca-suen committed:
fix: HttpUtilsTest generate the additional directory (#1310)

Co-authored-by: lucca <luccasuen.dev@gmail.com>

BigXin0109 committed:
Remove the threadpool directory under docker #1303  (#1315)

dependabot[bot] committed:
Bump junit from 4.12 to 4.13.1 in /agent (#1307)

Bumps [junit](https://github.com/junit-team/junit4) from 4.12 to 4.13.1.
- [Release notes](https://github.com/junit-team/junit4/releases)
- [Changelog](https://github.com/junit-team/junit4/blob/main/doc/ReleaseNotes4.12.md)
- [Commits](https://github.com/junit-team/junit4/compare/r4.12...r4.13.1)

---
updated-dependencies:
- dependency-name: junit:junit
  dependency-type: direct:development
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

HKMV committed:
Encrypted transmission of login password (#1316)

* feat(login): #1300 Encrypted transmission of login password

* feat(login): #1300 Modify the comments to English

* ref(login): #1300 Restore static resources

---------

Co-authored-by: limingwei <limingwei@century-cn.com>
Co-authored-by: Serenity <SerenitySir@outlook.com>

guanam committed:
add ClassUtilTest (#1318)

HKMV committed:
Extract the encryption section into a separate utility class (#1319)

* feat(login): #1300 Encrypted transmission of login password

* feat(login): #1300 Modify the comments to English

* ref(login): #1300 Restore static resources

* ref(console-login): Extract the encryption section into a separate utility class

---------

Co-authored-by: limingwei <limingwei@century-cn.com>
Co-authored-by: Serenity <SerenitySir@outlook.com>

magestacks committed:
The abstract thread pool dynamically changes the kernel layer (#1320)

Sakuragi27 committed:
test: add FastThreadPoolExecutorTest (#1317)

* test: add FastThreadPoolExecutorTest

* test: add FastThreadPoolExecutorTest

yhc777 committed:
check style  (#1299)

* style(hippo4j-threadpool-infra-common): checke style update

* style(hippo4j-threadpool-infra-common): checke style update

* style(hippo4j-threadpool-infra-common): check style log (info --> error)

* style(hippo4j-threadpool-infra-common): @Slf4j

* style(hippo4j-threadpool-infra-common): check style update

rnmb committed:
fix(console): fix examples login failed (#1321)

1. fix examples console login failed, because login tag is miss
2. repackage vue-element-admin into console static

magestacks committed:
Refactor the code to build the kernel layer (#1322)

magestacks committed:
Update hippo4j-agent-plugin version to revision (#1323)

yanrongzhen committed:
fix agent dist package (#1324)

magestacks committed:
Development Agent mode dynamic change (#1325)

weihubeats commented:
Please resolve conflicts

KaziAkibJaved committed:
resolved conflicts

KaziAkibJaved committed:
resolved conflicts

akib6075 commented:
Actually I changed these two files to resolve the flaky issue. So what else I can do to resolve this? Can you give me any clue? TIA.

weihubeats commented:
You can resolve conflicts by pulling the latest code from the remote develop branch and merging it into your branch

lucca-suen committed:
Refactor ThreadPoolInstanceRegistry to ThreadPoolRegistry. (#1339)

* Refactor ThreadPoolRegistry.

* Refactor ThreadPoolInstance Registry to ThreadPoolRegistry.

---------

Co-authored-by: lucca <luccasuen.dev@gmail.com>

akib6075 commented:
Thank you for your email

On Thu, 25 May, 2023 at 11:43 AM, weihubeats ***@***.***>
wrote:

> You can resolve conflicts by pulling the latest code from the remote
> develop branch and merging it into your branch
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/opengoofy/hippo4j/pull/1290#issuecomment-1562309362>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A6WBTKJGPLYWCXLFNPPOZA3XH3WRJANCNFSM6AAAAAAYDO36MY>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

akib6075 commented:
After doing so it's saying "Already up to date." No conflicts in my local after pulling remote develop branch.

rnmb committed:
feat(infra): add AbstractSubjectCenter test class #1338 (#1340)

* feat(infra): add AbstractSubjectCenter test class #1338
1. add SubjectCenter test class
2. add remove observer method
3. add get size for Observers method

Co-authored-by: rnmb<19276209+rnmb@users.noreply.github.com>

* feat(infra): add AbstractSubjectCenter test class
1. add SubjectCenter test class
2. rollback SubjectCenter method

Co-authored-by: rnmb<19276209+rnmb@users.noreply.github.com>

barret-yzh committed:
Add test case for ThreadFactoryBuilder (#1341)

* Create ThreadFactoryBuilderTest

* Update ThreadFactoryBuilderTest

Methods are covered by multiple unit test cases .
Console printing in English

* Update ThreadFactoryBuilderTest

Add open source protocol.
use @before annotation

* Update ThreadFactoryBuilderTest

* Update ThreadFactoryBuilderTest

* Update ThreadFactoryBuilderTest

rnmb committed:
feat(infra): add SyncPutQueuePolicyTest test class #1336 (#1344)

* feat(infra): add SyncPutQueuePolicyTest test class #1336
1. add SyncPutQueuePolicyTest test class

Co-authored-by: rnmb<19276209+rnmb@users.noreply.github.com>

* feat(infra): SyncPutQueuePolicyTest #1336
1. Add the open source agreement to the top, and leave a blank line with package

Co-authored-by: rnmb<19276209+rnmb@users.noreply.github.com>

annippolito committed:
test: Add test case for LogMessage #1331 (#1343)

* test: Add test case for LogMessage #1331

* change method signatures in LogMessage tests #1331

---------

Co-authored-by: annippolito <annibaleippolito@gmail.com>

lucca-suen committed:
Add test case for RejectedPolicyTypeEnum (#1345)

* test: Add test case for RejectedPolicyTypeEnum

* refactor: Add open source protocol for RejectedPolicyTypeEnumTest.

---------

Co-authored-by: lucca <luccasuen.dev@gmail.com>

barret-yzh committed:
Add test case for MessageWrapper (#1346)

* Create ThreadFactoryBuilderTest

* Update ThreadFactoryBuilderTest

Methods are covered by multiple unit test cases .
Console printing in English

* Update ThreadFactoryBuilderTest

Add open source protocol.
use @before annotation

* Update ThreadFactoryBuilderTest

* Update ThreadFactoryBuilderTest

* Update ThreadFactoryBuilderTest

* Add test case for MessageWrapper

* Update MessageWrapperTest

annippolito committed:
test: Add test case for RunsOldestTaskPolicy #1335 (#1347)

* test: Add test case for RunsOldestTaskPolicy #1335

* make test viewable by codecov #1335

annippolito committed:
fix: increase codecov coverage by using junit instead of junit.jupiter (#1349)

* increase codecov coverage by using junit instead of junit.jupiter

* increase codecov coverage in some other packages

* remove blank line

lucca-suen committed:
Add test case for JacksonHandler. (#1353)

* test: Add test case for JacksonHandler.

* refactor: Set permission modifiers for JacksonHandlerTest's classes and methods.

---------

Co-authored-by: lucca <luccasuen.dev@gmail.com>

Mryanhehe committed:
test ResizableCapacityLinkedBlockingQueue (#1350)

shuhai65 committed:
Add test case for IoUtil (#1352)

* Add test case for IoUtil

* replace import java.io.*;

* 1.change testTryDecompress to testTryDecompressInputStream and testTryDecompressByteArray
2.remove test resource file

DDreame committed:
add UnitTest about MessageConvert (#1358)

* add UnitTest about MessageConvert

* add UnitTest about MessageConvert

* add UnitTest about MessageConvert

furaul committed:
prototype version of dubbo threadpool agent. (#1357)

* prototype version of dubbo threadpool agent.

* rename to optimze & fix ci errors.

* rename to optimze & fix ci errors.

magestacks committed:
Update using company (#1359)

barret-yzh committed:
Add test case for ThreadPoolExecutorUtil  (#1356)

* Create ThreadFactoryBuilderTest

* Update ThreadFactoryBuilderTest

Methods are covered by multiple unit test cases .
Console printing in English

* Update ThreadFactoryBuilderTest

Add open source protocol.
use @before annotation

* Update ThreadFactoryBuilderTest

* Update ThreadFactoryBuilderTest

* Update ThreadFactoryBuilderTest

* Add test case for MessageWrapper

* Update MessageWrapperTest

* Add test case for ThreadPoolExecutorUtil

Add test case for ThreadPoolExecutorUtil

* Update ThreadPoolExecutorUtilTest

yhc777 committed:
feature(issue 1272): FeiShu group chat robot adds signature verification function (#1360)

* feature(issue 1272): FeiShu group chat robot adds signature verification function

* fix(#1272): 1、sign is null fix 2、code reformat

* refactor(#1272): sign refactor

BigXin0109 committed:
Docker supports arm64 images #1279 (#1367)

* feat: docker support linux/arm64,linux/amd64 platform

* feat: docker support linux/arm64,linux/amd64 platform

magestacks committed:
Abstract alarm and notification kernel modules and refactor existing code (#1376)

magestacks committed:
Fix alarm notification asynchronous thread pool factory error (#1377)

zjHe committed:
@EnableDynamicThreadPool controls MicrometerMonitorAutoConfiguration (#1380)

yanrongzhen committed:
Bruceyan/issue monitor 20230624 (#1382)

wo883721 committed:
#1251 Resolved two registration issues (#1383)

Co-authored-by: 章新好 Sunny Zhang <sunny.zhang@greaconsulting.com>

MagikLau committed:
Update AdaptedThreadPoolDestroyPostProcessor.java add default impl (#1384)

Fix match case: spring-beans-4.3.21.RELEASE.jar
Forked from: spring-beans-5.1.14.RELEASE.jar

magestacks committed:
Update README.md (#1398)

magestacks committed:
Update doc (#1399)

* Update docs

* Update users.md

shizi327 committed:
Refactoring throws an exception when the newly set number of core threads exceeds the maximum number of threads (#1409)

Co-authored-by: 2590965087@qq.com <2590968087@qq.com>

LukeLiang committed:
Replace the buffer pool type from ArrayBlockingQueue to LinkedBlockin… (#1411)

* Replace the buffer pool type from ArrayBlockingQueue to LinkedBlockingQueue type

* Import LinkedBlockingQueue package

Han-Xinlong committed:
Service exception test (#1410)

* hxl- commit - 7-29

* Service exception test

* Service exception test update

zjHe committed:
@EnableDynamicThreadPool controls MicrometerMonitorAutoConfiguration (#1413)

GRL-bxy committed:
fix：create thread poll content not change (#1416)

Pan-YuJie committed:
Feature: server add Ldap user authentication (#1392)

GRL-bxy committed:
Feat/new UI (#1422)

* Remove console autofill hippo4j username and password logic (#1246)

* feat:更改网址

* feat: submit new-ui

yanrongzhen committed:
Add some issue templates. (#1430)

* add some issue templates.

* update develop guidelines link.

yanrongzhen committed:
Modify label names in ISSUE templates. (#1431)

magestacks committed:
Update users.md

iwangjie committed:
Merge pull request #1432 from magestacks/docs_20230831_users

Update users.md

yanrongzhen committed:
Remove unnecessary agent config items. (#1435)

pandalee99 committed:
Remove the redundant comment for CalculateUtil (#1437)

2084276225 committed:
feature: hippo4j config model in nacos config before check (#1438)

liguangling-lgl committed:
feature:check the required fields etcd when the Config mode starts(#1439) (#1440)

magestacks committed:
Spotless code formatting (#1442)

spongebody committed:
Introduce the resources required for website optimization (#1444)

spongebody committed:
Initial homepage development and multilingual settings (#1445)

Pan-YuJie committed:
Fix Server-example Unknown user (#1446)

* Feature: server add Ldap user authentication

* Fix Server-example Unknown user

* Remove RFC7230Config.class

bobowiki committed:
feat:tidy up front frame (#1449)

Co-authored-by: yikai <yikai@didiglobal.com>

spongebody committed:
Initial completion of user page development (#1450)

Myboolean committed:
check the required fields apollo when the Config mode starts (#1452)

dmego committed:
fix: user rights module clicks the add button without loading the tenant data (#1451)

bobowiki committed:
feat: Tidy up front end layout (#1454)

Co-authored-by: yikai <yikai@didiglobal.com>

bobowiki committed:
feat: Tidy directory structure (#1456)

Co-authored-by: yikai <yikai@didiglobal.com>

GRL-bxy committed:
Feature/new UI (#1459)

* Remove console autofill hippo4j username and password logic (#1246)

* feat:login + wrapper request

bobowiki committed:
request by fetch utils done  (#1461)

* fix: 合并代码

* feat: requst utils done

---------

Co-authored-by: yikai <yikai@didiglobal.com>

spongebody committed:
Homepage layout adjustment, and adding new image resource (#1462)

yanrongzhen committed:
ignore DS_Store file (#1463)

furaul committed:
Added Undertow thread pool parameters (#1464)

bobowiki committed:
Add i18n header components (#1466)

baiJinjin committed:
Add tenant management

bobowiki committed:
Merge pull request #1467 from baiJinjin/feature_htl

Add tenant management

bobowiki committed:
Feat oe thread pool (#1468)

* feat: thread pool

* feat: remove yarnloack

---------

Co-authored-by: yikai <yikai@didiglobal.com>

xjlgod committed:
feature: hippo4j config model in zookeeper config before check (#1457)

baiJinjin committed:
Feature htl (#1471)

* Add user log item Manage

* edit Item Management

* merge from develop

* add router

bobowiki committed:
Feat login page (#1473)

* feat: login

* feat: recommmit remove node-module

---------

Co-authored-by: yikai <yikai@didiglobal.com>

baiJinjin committed:
add router and edit request (#1477)

* Add user log item Manage

* edit Item Management

* merge from develop

* add router

* add router

* edit logmanage

* edit tenant

baiJinjin committed:
perfect the user log tenant and project modules (#1478)

* Add user log item Manage

* edit Item Management

* merge from develop

* add router

* add router

* edit logmanage

* edit tenant

* update item

* update user tenant log user

* edit userName

bobowiki committed:
feat: login en and happy work (#1479)

* stash code

* fix active router

* feat: login en and happy work

---------

Co-authored-by: yikai <yikai@didiglobal.com>

spongebody committed:
Page adjustment and optimization (#1480)

GRL-bxy committed:
feat:add_global_tenant + delete_search_tenant (#1482)

Co-authored-by: 吹泡泡的团子 <baoxinyi_i@didiglobal.com>

bobowiki committed:
feat: fix login encrypt and prettier .vscode (#1481)

Co-authored-by: yikai <yikai@didiglobal.com>

spongebody committed:
Modify the active color of the document and adjust the background of the hero (#1483)

magestacks committed:
Modify front-end text description

weihubeats committed:
Merge pull request #1484 from magestacks/docs_20230927

Modify front-end text description

spongebody committed:
Make detailed adjustments according to the document. (#1485)

GRL-bxy committed:
feat: delete select tenent and Inject globalTenant (#1487)

Co-authored-by: 吹泡泡的团子 <baoxinyi_i@didiglobal.com>

GRL-bxy committed:
feat:add thread pool (#1488)

Co-authored-by: 吹泡泡的团子 <baoxinyi_i@didiglobal.com>

magestacks committed:
Develop thread pool monitoring capabilities (#1489)

valery1707 committed:
Check for null value with the appropriate assertion method (#1493)

bbelide2 committed:
Use JSONAssert instead of Assert in JSONUtilTest (#1495)

Co-authored-by: bbelide2 <bbelide2@fa23-cs527-002.cs.illinois.edu>

bbelide2 committed:
Fix flaky test in Md5UtilTest (#1494)

* Fix flaky test in Md5UtilTest

* Remove unused imports

---------

Co-authored-by: bbelide2 <bbelide2@fa23-cs527-002.cs.illinois.edu>

mazengrun committed:
check the required fields of polaris when the config mode starts (#1470)

* check the required fields of polaris when the config mode starts

* Modify the format of polaris in BootstrapConfigProperties

* Define static constants of polaris in BeforeCheckConfiguration

* Use configuration and constants exist in  PolarisRefresherHandler

* Just consider the properties of BootstrapPropertiesInterface

* Don't throw exception in if else

* Optimize code format in BeforeCheckConfiguration

bbelide2 committed:
Fix flaky tests in LogMessageTest (#1500)

* Fix flaky tests in LogMessageTest

* Remove fix for NOD test

---------

Co-authored-by: balasukesh <bbelide2@fa23-cs527-002.cs.illinois.edu>

bbelide2 committed:
Fix flaky test in HttpUtilsTest (#1499)

Co-authored-by: balasukesh <bbelide2@fa23-cs527-002.cs.illinois.edu>

KaziAkibJaved committed:
Issue fixed for assertGetPoolContent method

KaziAkibJaved committed:
Issue fixed for assertGetPoolContent method

KaziAkibJaved committed:
Merge branch 'new-fix-flaky' of github.com:akib6075/hippo4j into new-fix-flaky

akib6075 commented:
@weihubeats there is a loophole in the assertGetPoolContent() which is the testText is a collection of threadpoolparameter info which has a specific order of the keys. But the ContentUtil.getPoolContent() returns the threadpoolparameter info having different order of keys. So it's happening because of different order of keys in the Json serialization.
This is failed CI log: https://github.com/akib6075/hippo4j/actions/runs/5135847635

To fix this I made a solution which serializes and deserializes to check that the object is the same. Now the question can come that my solution uses ContentUtil.getPoolContent() twice. So the result must throw error again. But the answer is no. Because the caching of properties is implemented in the databind module's internal classes, such as com.fasterxml.jackson.databind.introspect.POJOPropertiesCollector and com.fasterxml.jackson.databind.introspect.POJOPropertyBuilder. These classes are responsible for inspecting the structure of Java classes and extracting their properties.
Jackson utilizes a caching mechanism for property introspection, which involves caching the metadata associated with Java classes to improve performance during serialization and deserialization. This caching is performed by the Jackson databind module, which is typically used in conjunction with the ObjectMapper.
JacksonHandler.java used this library by importing com.fasterxml.jackson.databind.*;
The fixed CI log: https://github.com/akib6075/hippo4j/actions/runs/5190631210

weihubeats commented:
You can mention an issue in the issues list, and then associate it here, see this pr: https://github.com/opengoofy/hippo4j/pull/1380

weihubeats commented:
Why do you have so many file changes? I feel like there should be something wrong with your code merge? Wouldn't it be a good idea to consider pulling a new branch from the latest branch and re-committing your code

akib6075 commented:
@weihubeats Thank you for your feedback. I understand your concern about the large number of file changes. I've reviewed the commits and believe they are necessary to address the flaky tests issue. However, I'm open to discussing alternative approaches. Would you like me to provide more details about the changes or would you prefer I rebase the branch and submit a new merge request? Also I've opened same PR which is [#1373](https://github.com/opengoofy/hippo4j/pull/1373). You can check that as well.
